pkg/controller/storagecluster/reconcile.go

```diff
@@ -406,6 +407,11 @@ func (r *ReconcileStorageCluster) validateStorageDeviceSets(sc *ocsv1.StorageClu
 				return fmt.Errorf("failed to validate StorageDeviceSet %d: no StorageClass specified for metadataPVCTemplate", i)
 			}
 		}
+		if ds.DeviceType != "" {
+			if strings.EqualFold(ds.DeviceType, "SSD") || strings.EqualFold(ds.DeviceType, "HDD") || strings.EqualFold(ds.DeviceType, "NVMe") {
```

Comment:
It is good to define these values as types and use it here

Reply:
can you explain more what types and how can we ensure to ignore case insensitivity.

Reply:
something like this https://github.com/openshift/ocs-operator/blob/master/pkg/controller/storagecluster/uninstall_reconciler.go#L24-L43

Reply:
why would I want to use this approach. when I want to use devicetype only once in file. it is not like `ssd` or `ssd` is being used more than once.

Reply:
It was just a suggestion and totally up to you.

Reply:
I would make it a demand. :) It is good practice to use explicit types rather than just comparing raw strings. You could then also create an array of these types, and just search to see if the array contains the given value.

Reply:
the main concern is to check for case sensitivity. https://github.com/openshift/ocs-operator/pull/859#discussion_r513376637